lib/language.js

```diff
@@ -0,0 +1,113 @@
+'use strict';
+
+// Load modules
+
+var Boom = require('boom');
+var Hoek = require('hoek');
+
+
+// Declare internals
+
+var internals = {};
+
+
+// https://tools.ietf.org/html/rfc7231#section-5.3.5
+// Accept-Language: da, en-gb;q=0.8, en;q=0.7
+
+
+exports.language = function (header, preferences) {
+
+    Hoek.assert(!preferences || Array.isArray(preferences), 'Preferences must be an array');
+    var languages = exports.languages(header);
+
+    if (languages.length === 0) {
+        languages.push('');
+    }
+
+    // No preferences.  Take the first charset.
+
+    if (!preferences || preferences.length === 0) {
+        return languages[0];
+    }
+
+    // If languages includes * return first preference
+
+    if (~languages.indexOf('*')) {
+        return preferences[0];
+    }
+
+    // Try to find the first match in the array of preferences
+
+    preferences = preferences.map(function (str) {
```

Comment:
Could be.  I've seen some old perf statements indicating a big difference, but nothing that's recent.  My gut says there probably isn't a huge difference now days, but I haven't done anything to bear that out.  For now I'll stick with the current map implementation, and make an issue to determine what needs to be done with that.